File: android-app/app/src/main/java/com/shinomeno/taktak/BootCompletedReceiver.java
package com.shinomeno.taktak;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;

public final class BootCompletedReceiver extends BroadcastReceiver {
    @Override
    public void onReceive(Context context, Intent intent) {
        if (intent != null && Intent.ACTION_BOOT_COMPLETED.equals(intent.getAction())) {
            BackgroundRealtimeService.startIfAllowed(context.getApplicationContext());
        }
    }
}

File: android/shno-mano-call/app/src/main/java/com/shnomano/call/BootCompletedReceiver.kt
package com.shnomano.call

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent

class BootCompletedReceiver : BroadcastReceiver() {
    override fun onReceive(context: Context, intent: Intent?) {
        if (intent?.action != Intent.ACTION_BOOT_COMPLETED) return
        if (com.shnomano.call.data.SessionStore(context).isSignedIn()) {
            BackgroundRealtimeService.start(context)
        }
    }
}
